firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignmentListTest.java

```diff
@@ -0,0 +1,137 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.metadata;
+
+import com.google.common.truth.Truth;
+import com.google.firebase.crashlytics.internal.CrashlyticsTestCase;
+import com.google.firebase.encoders.DataEncoder;
+import com.google.firebase.encoders.json.JsonDataEncoderBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import org.json.JSONException;
+import org.junit.Test;
+
+public class RolloutAssignmentListTest extends CrashlyticsTestCase {
+  private static final String ROLLOUTS_STATE_JSON_1 =
+      "[{"
+          + "\"rolloutId\":\"rollout_1\","
+          + "\"variantId\":\"control\","
+          + "\"parameterKey\":\"my_feature\","
+          + "\"parameterValue\":\"false\","
+          + "\"templateVersion\":1"
+          + "}]";
+
+  private static final String ROLLOUTS_STATE_JSON_2 =
+      "["
+          + "{"
+          + "\"rolloutId\":\"rollout_1\","
+          + "\"variantId\":\"control\","
+          + "\"parameterKey\":\"my_feature\","
+          + "\"parameterValue\":\"false\","
+          + "\"templateVersion\":1"
+          + "},"
+          + "{"
+          + "\"rollout_id\":\"rollout_2\""
+          + "\"variantId\":\"control\","
+          + "\"parameterKey\":\"color_feature\","
+          + "\"parameterValue\":\"blue\","
+          + "\"templateVersion\":2"
+          + "}]";
+
+  private static List<RolloutAssignment> ROLLOUTS_STATE_1;
+
+  private static List<RolloutAssignment> ROLLOUTS_STATE_2;
+
+  private static final String ROLLOUT_ASSIGNMENT_JSON =
+      "{"
+          + "\"rolloutId\":\"rollout_1\","
+          + "\"variantId\":\"control\","
+          + "\"parameterKey\":\"my_feature\","
+          + "\"parameterValue\":\"false\","
+          + "\"templateVersion\":1"
+          + "}";
+
+  private static final int MAX_ENTRIES = 64;
+
+  private final RolloutAssignmentList rolloutAssignmentList =
+      new RolloutAssignmentList(MAX_ENTRIES);
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+
+    ROLLOUTS_STATE_1 = new ArrayList<>();
+    ROLLOUTS_STATE_1.add(
+        RolloutAssignment.create("rollout_1", "my_feature", "false", "control", 1));
+
+    ROLLOUTS_STATE_2 = new ArrayList<>();
+    ROLLOUTS_STATE_2.add(
+        RolloutAssignment.create("rollout_1", "my_feature", "false", "control", 1));
+    ROLLOUTS_STATE_2.add(
+        RolloutAssignment.create("rollout_2", "color_feature", "false", "control", 2));
+  }
+
+  @Test
+  public void testRollAssignmentInit_json() throws JSONException {
+    RolloutAssignment expected =
+        RolloutAssignment.create("rollout_1", "my_feature", "false", "control", 1);
+    RolloutAssignment rolloutAssignmentInitFromJson =
+        RolloutAssignment.create(ROLLOUT_ASSIGNMENT_JSON);
+
+    Truth.assertThat(rolloutAssignmentInitFromJson).isEqualTo(expected);
```

Comment:
Nit: If you static import `Truth.assertThat` these will look better. It's just a style thing.